### PR TITLE
Updated node/ruby version requirements

### PR DESCRIPTION
**Pull request by shleeable**
Hey, nodejs v8 and Ruby 2.4 should be considered EOL. So I've bumped them up to more recent releases.

Node v13 / Ruby 2.7 will be stable soon but are untested.

This has the benefit of allowing blocked PRs #12521 #11727 #12434 etc

### CODE REVIEW

**Comment by Shleeble** [comment]
lol. Forgot to delete the CI.

**Comment by Shleeble** [comment]
Ready to merge.

**Comment by Shleeble** [comment]
Please review @ThibG @trwnh @ykzts @Gargron.

File: README.md [hunk 1]
```diff
@@ -65,12 +65,12 @@ Mastodon acts as an OAuth2 provider so 3rd party apps can use the REST and Strea
...
+**Requirements: (Recommended)**
 
-- **PostgreSQL** 9.5+
-- **Redis**
-- **Ruby** 2.4+
-- **Node.js** 8+
+- **PostgreSQL** 9.5+ (10.6+)
+- **Redis** 4+ (5+)
+- **Ruby** 2.5+ (2.6+)
+- **Node.js** 10.13+ (12.13+)
```
Comment (ThibG) [inline README.md]: Is there a specific reason for those recommendations?

**Review by ClearlyClaire — COMMENTED** [review]
Seems fine, but I'm not sure there is any rush yet. Node 8's EOL is at the end of the month, but Ruby 2.4's EOL is in a few more months.

I see no strong reason to reject this change, but it can probably wait after the next Mastodon release (assuming there's one shortly, which is what I'm advocating for)

File: README.md (hunk 1 above)
Comment (Shleeble) [reply README.md]: Ruby 2.7 will be out in a month or so - That'll likely be around the time of the EOL for the others too.

and the recommendations are just the versions used in Docker - so they're tested and slightly more bleeding edge.

**Comment by Shleeble** [comment]
FYI Dropping Ruby 2.4 unblocks #12584 
"sprockets-4.0.0 requires ruby version >= 2.5.0"

File: README.md (hunk 1 above)
Comment (ThibG) [reply README.md]: ok, recommending the versions used in the Docker image seem reasonable to me (but maybe that reason should be explicitly written there as well), I'm just not sure it's worth the burden of having to maintain that information when such versions change

File: package.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "name": "@tootsuite/mastodon",
   "license": "AGPL-3.0-or-later",
   "engines": {
-    "node": ">=8.12 <13"
+    "node": ">=10.13 <13"
```
Comment (abcang) [inline package.json]: The reason I set the lower limit of node version to 8.12 is to match the lower limit version of the package being used. I think it is better to set it to 10 if there is no special reason to specify the minor version.

Comment (Shleeble) [reply package.json]: Similar reasons for mine. #12521 is 10.13 for that package.

Comment (abcang) [reply package.json]: ok, I understand.

File: README.md (hunk 1 above)
Comment (SuperSandro2000) [reply README.md]: I think it should be written out, too. It is quite confusing if you look at it the first time.

**Comment by Shleeble** [comment]
I'll rejig this in the next few weeks.

**Pull request closed without merging**